### DISCUSSION TITLE
Getter method for Titanium modules versions

### DISCUSSION THREAD

**Discussion by hbugdoll** [appcelerator/titanium_mobile#13553] [General]
To obtain the Titanium SDK version at runtime, you can simply use `Titanium.version`.
It would be helpful to also get the version of a **Titanium module** programmatically, e.g. `MyModule.version`.

**Comment by m1ga** [comment]
I'm not sure if there is an automated way in doing this, but you can add this to your module
```
@Kroll.getProperty
    public String version() {
        return BuildConfig.TI_MODULE_VERSION;
    }
``` 
and then you can use `MyModule.version`

When you build your module .e.g Firebase-FCM there is a BuildConfig.java file in `android/build/module/build/generated/source/buildConfig/release/firebase/cloudmessaging/`. That contains strings like the version, module name, author,... You can access them as shown above. 

But is it interesting to know the version of a module at runtime? It should be connected to your app version. Same as Titanium version. If I know my app version I can lookup modules and Ti SDK version.

**Reply by hbugdoll** [reply]
Thanks @m1ga for your quick response and suggestions.

> But is it interesting to know the version of a module at runtime?

E.g. to simply display it to the user in a legal notice resp. credits (integrated software section).
The reasons are similar to the existence of `Titanium.App.version`.

**Reply by m1ga** [reply]
Ah, good point :+1: Never thought about automating that but interesting idea. Could be added to the default module preset so newer modules will have that but you can't really rely on it since not every module has it and the dev could remove it by hand.
